src/include/sof/schedule.h

```diff
@@ -25,113 +25,56 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  *
- * Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
+ * Author: Bartosz Kokoszko <bartoszx.kokoszko@linux.intel.com>
  */
 
-#ifndef __INCLUDE_SOF_SCHEDULE_H__
-#define __INCLUDE_SOF_SCHEDULE_H__
+/* Generic schedule api header */
+#ifndef __INCLUDE_SOF_SCHEDULER_H__
+#define __INCLUDE_SOF_SCHEDULER_H__
 
 #include <stdint.h>
-#include <stddef.h>
-#include <errno.h>
-#include <sof/sof.h>
-#include <sof/lock.h>
 #include <sof/list.h>
-#include <sof/work.h>
-#include <sof/wait.h>
-
-/* schedule tracing */
-#define trace_schedule(format, ...) \
-	trace_event(TRACE_CLASS_SCHEDULE, format, ##__VA_ARGS__)
-
-#define trace_schedule_error(format, ...) \
-	trace_error(TRACE_CLASS_SCHEDULE, format, ##__VA_ARGS__)
-
-#define tracev_schedule(format, ...) \
-	tracev_event(TRACE_CLASS_SCHEDULE, format, ##__VA_ARGS__)
-
-struct schedule_data;
-struct sof;
-
-/* task states */
-#define TASK_STATE_INIT		0	
-#define TASK_STATE_QUEUED	1
-#define TASK_STATE_PENDING	2
-#define TASK_STATE_RUNNING	3
-#define TASK_STATE_PREEMPTED	4
-#define TASK_STATE_COMPLETED	5
-#define TASK_STATE_FREE		6
-#define TASK_STATE_CANCEL	7
-
-/* task priorities - values same as Linux processes, gives scope for future.*/
-#define TASK_PRI_LOW	19
-#define TASK_PRI_MED	0
-#define TASK_PRI_HIGH	-20
-
-#define TASK_PRI_IPC	1
-#define TASK_PRI_IDC	1
-
-/* maximun task time slice in microseconds */
-#define SCHEDULE_TASK_MAX_TIME_SLICE	5000
-
-/* task descriptor */
-struct task {
-	uint16_t core;			/* core id to run on */
-	int16_t priority;		/* scheduling priority TASK_PRI_ */
-	uint64_t start;			/* scheduling earliest start time */
-	uint64_t deadline;		/* scheduling deadline */
-	uint32_t state;			/* TASK_STATE_ */
-	struct list_item list;		/* list in scheduler */
-	struct list_item irq_list;	/* list for assigned irq level */
-
-	/* task function and private data */
-	void *data;
-	void (*func)(void *arg);
 
-	/* runtime duration in scheduling clock base */
-	uint64_t max_rtime;		/* max time taken to run */
-	completion_t complete;
-};
+#define SOF_EDF_TASK		0
+#define SOF_LOW_LATENCY_TASK	1
 
-struct schedule_data **arch_schedule_get(void);
+#define SOF_TASK_LOW	0
+#define SOF_TASK_MED	1
+#define SOF_TASK_HIGH	2
 
-void schedule(void);
+#define SOF_TASK_INIT_STATE	0
 
-void schedule_task(struct task *task, uint64_t start, uint64_t deadline);
+/* Scheduler flags */
+#define SOF_SCHEDULE_IDLE  (1 << 0)
 
-void schedule_task_idle(struct task *task, uint64_t deadline);
+extern uint32_t task_priority[2][3];
 
-int schedule_task_cancel(struct task *task);
+struct sof_task {
```

Comment:
Can we go just with task without sof or at least sched_task?
I would also drop sof from any schedule method names.